### PR TITLE
Handling HTML error messages

### PR DESCRIPTION
**Pull request by alexbourret**
Story details: https://app.shortcut.com/dataiku/story/94528

### CODE REVIEW

File: python-lib/common.py [hunk 1]
```diff
@@ -68,13 +68,25 @@ def is_request_performed(response):
...
         seconds_before_retry = decode_retry_after_header(response)
         logger.warning("Sleeping for {} seconds".format(seconds_before_retry))
         time.sleep(seconds_before_retry)
         return False
+    if not is_response_json(response):
+        logger.error("JSON is expected but SharePoint Online response is something else")
+        logger.error("on url {}, dump={}, server headers={}".format(response.url, response.content, response.headers))
+        seconds_before_retry = 20
+        logger.warning("Sleeping for {} seconds".format(seconds_before_retry))
+        time.sleep(seconds_before_retry)
```
[the remark this answers is not in the document]
Comment (liamlynch-data) [reply python-lib/common.py:82]: Discussed - this case shouldn't return false, in fact, but should throw an exception which is caught higher up before the JSON parsing will fail and cause a crash.
(Sleeping is also not necessary)

[the remark this answers is not in the document]
Comment (liamlynch-data) [reply python-lib/common.py:82]: And we can test the exception is thrown in test_common.py

**Review by liamlynch-data — CHANGES_REQUESTED** [review]
There's a change to make as we discussed. Otherwise looks good (one other minor point below)

File: python-lib/common.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-lib/common.py]: After trying this approach, it appears that raising exception on non json pages results in creating a new set of issues. So I removed this part of the code in [eef49be](https://github.com/dataiku/dss-plugin-sharepoint-online/pull/40/commits/eef49bef27aa94c169b3ea82ed8cab38636ae52e)

File: python-lib/common.py [hunk 2]
```diff
@@ -68,13 +68,25 @@ def is_request_performed(response):
         return False
     if response.status_code in [429, 503]:
         logger.warning("Error {}, headers = {}".format(response.status_code, response.headers))
+        if response.status_code == 503:
+            logger.warning("dumping content: {}".format(response.content))
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-lib/common.py]: 👍  Done in [d7b6809](https://github.com/dataiku/dss-plugin-sharepoint-online/pull/40/commits/d7b680962b55b6f066c3a621a5fc13524d1e7ee8)

File: tests/python/unit/test_common.py [hunk 3]
```diff
@@ -23,7 +25,7 @@ def setup_class(self):
         self.ok_path_2 = ["a", "d"]
         self.ko_path = ["a", "c"]
         self.mock_response_none = None
-        self.mock_response_http_200 = MockResponse(200, {"Retry-After": "1"})
+        self.mock_response_http_200 = MockResponse(200, {"Content-Type": "application/json", "Retry-After": "1"})
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply tests/python/unit/test_common.py]: 👍  done in [c217142](https://github.com/dataiku/dss-plugin-sharepoint-online/pull/40/commits/c2171428e45bd9227517be826e23caa8b1e484a7)

[the remark this answers is not in the document]
Comment (liamlynch-data) [reply tests/python/unit/test_common.py]: Great - thanks!

File: python-lib/common.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (liamlynch-data) [reply python-lib/common.py]: Looks like the best soln, all good from my POV

**Pull request merged**